src/Sarif.Multitool/Rules/SARIF2005.ProvideToolProperties.cs

```diff
@@ -74,17 +105,38 @@ private void AnalyzeToolDriver(ToolComponent toolComponent, string toolDriverPoi
                 }
             }
 
-            if (string.IsNullOrWhiteSpace(toolComponent.Version) && string.IsNullOrWhiteSpace(toolComponent.SemanticVersion))
+            bool informationUriRequired = this.Context.Policy.GetProperty(InformationUriRequired);
+            if (informationUriRequired && toolComponent.InformationUri == null)
             {
-                // {0}: The tool '{1}' provides neither a 'version' property nor a 'semanticVersion'
-                // property. Providing a version enables the log file consumer to determine whether
-                // the file was produced by an up to date version, and to avoid accidentally comparing
-                // log files produced by different tool versions.
+                // {0}: The tool '{1}' does not provide 'informationUri'. This property helps the
+                // developer responsible for addessing a result by providing a way to learn more
+                // about the tool.
                 LogResult(
                     toolDriverPointer,
-                    nameof(RuleResources.SARIF2005_ProvideToolProperties_Warning_ProvideToolVersion_Text),
+                    nameof(RuleResources.SARIF2005_ProvideToolProperties_Warning_ProvideToolnformationUri_Text),
                     toolComponent.Name);
             }
+
+            StringSet acceptableVersionProperties = this.Context.Policy.GetProperty(AcceptableVersionProperties);
+            bool toolDriverProvidesVersion =
+                (acceptableVersionProperties.Contains(nameof(toolComponent.Version)) && !string.IsNullOrWhiteSpace(toolComponent.Version))
+                ||
+                (acceptableVersionProperties.Contains(nameof(toolComponent.SemanticVersion)) && !string.IsNullOrWhiteSpace(toolComponent.SemanticVersion))
+                ||
+                (acceptableVersionProperties.Contains(nameof(toolComponent.DottedQuadFileVersion)) && !string.IsNullOrWhiteSpace(toolComponent.DottedQuadFileVersion));
```

Comment:
please take another crack at making this code readable. I can appreciate that it is a challenge. Consider using the |= in a sequence to help out. We don't really care that this would be less optimal (losing benefit of short-circuiting).

bool blah = SomeComputation(baz);
blah |= AnotherComputation(bar);
blah |= Etc(biz);

If (blah) Hooray();

Reply:
Changed to sequential `|=` but IMO the original was already beautifully readable. YMMV.

---
In reply to: [478507288](https://github.com/microsoft/sarif-sdk/pull/2044#discussion_r478507288) [](ancestors = 478507288)

Reply:
My mileage did vary, I did not find it readable. Thanks for being willing to change.

---
In reply to: [478521594](https://github.com/microsoft/sarif-sdk/pull/2044#discussion_r478521594) [](ancestors = 478521594,478507288)